pytorch_lightning/profiler/pytorch.py

```diff
@@ -519,11 +491,14 @@ def _delete_profilers(self) -> None:
             self._register.__exit__(None, None, None)
             self._register = None
 
+    def _clear_recording_map(self) -> None:
```

Comment:
how about https://stackoverflow.com/questions/5384914/how-to-delete-items-from-a-dictionary-while-iterating-over-it?